### PR TITLE
C2t2 validate event search

### PR DESCRIPTION
**Pull request by georginaw**
Created front end (JS and HTML tag) and back end validation (in the get events controller invoke function) to check if the user has put in a value in the input and that it's not longer than 255 characters

### CODE REVIEW

File: public/js/eventsPage.js [hunk 1]
```diff
@@ -8,8 +8,13 @@ const message = document.querySelector('#messages')
  *
  * @return event data
  */
-function getEvents() {
-    fetch('./api/getEvents', {
+function getEvents(search = false) {
+    let url = './api/getEvents'
+    if (search !== false) {
```
Comment (anton25360) [inline public/js/eventsPage.js]: Why did you use !== false rather than == true?.

File: public/js/eventsPage.js [hunk 2]
```diff
@@ -180,4 +185,15 @@ function validateForm() {
...
     return success
-}
\ No newline at end of file
+}
+
+
+document.querySelector('#submit-search-event').addEventListener('click', function(e) {
+    const searchInput = document.querySelector('#academy-events-search').value
+    e.preventDefault()
+    if ((searchInput.length) && searchInput.length < 255) {
```
Comment (anton25360) [inline public/js/eventsPage.js]: Check '255'. It should probably be 256, just like in this line:
`if (strlen($eventSearchInput) < 256) {`

File: public/js/eventsPage.js (hunk 1 above)
Comment (georginaw) [reply public/js/eventsPage.js]: This is a strict comparison and makes more sense than using true (used when the input it blank)

File: public/js/eventsPage.js (hunk 2 above)
Comment (georginaw) [reply public/js/eventsPage.js]: Updated to 256

**Pull request merged**